pkg/cmd/app.go

```diff
@@ -397,17 +397,27 @@ func enableDCGMExpXIDErrorsCountCollector(cs *dcgmexporter.CounterSet, fieldEnti
 }
 
 func getFieldEntityGroupTypeSystemInfo(cs *dcgmexporter.CounterSet, config *dcgmexporter.Config) *dcgmexporter.FieldEntityGroupTypeSystemInfo {
-	allCounters := []dcgmexporter.Counter{}
+	var allCounters []dcgmexporter.Counter
 
 	allCounters = append(allCounters, cs.DCGMCounters...)
-	allCounters = append(allCounters,
-		dcgmexporter.Counter{
-			FieldID: dcgm.DCGM_FI_DEV_CLOCK_THROTTLE_REASONS,
-		},
-		dcgmexporter.Counter{
-			FieldID: dcgm.DCGM_FI_DEV_XID_ERRORS,
-		},
-	)
+
+	if len(cs.ExporterCounters) > 0 {
```

Comment:
I would prefer either a comment about why these fields need special handling or put the special handling in an appropriately named separate function.